File: c++/6 kyu Rectangle into Squares.cpp
#include <vector>
using namespace std;
class SqInRect
{
  public:
  static std::vector<int> sqInRect(int lng, int wdth){
  if(lng==wdth)return {};
	vector<int>c;
  while(lng!=0 && wdth!=0)		
		if(lng>wdth){c.push_back(wdth); lng=lng-wdth;}
		else   {c.push_back(lng); wdth=wdth-lng;}
	return c;
  }
};

File: Kotlin/6 kyu Rectangle into Squares.kt
package squares

fun sqInRect(lng:Int, wdth:Int):List<Int>? {
	var a=lng
	var b=wdth
	if(a==b)return null
	var c:List<Int> = listOf()
    while(a!=0 && b!=0)		
		if(a>b){c+=listOf(b); a=a-b;}
		else   {c+=listOf(a); b=b-a;}
	return c
}
